### PR TITLE
Switch from gnutls to glib for MD5 hashing

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

**Comment by clumens** [comment]
I could additionally revert the patches that made gnutls mandatory.

File: lib/common/digest.c [hunk 1]
```diff
@@ -226,37 +226,22 @@ pcmk__xa_filterable(const char *name)
...
 
-    digest = pcmk__assert_alloc(1 + (2 * dlen), sizeof(char));
-
-    for (int i = 0; i < dlen; i++) {
-        sprintf(digest + (2 * i), "%02x", raw_digest[i]);
-    }
-
-    digest[(2 * dlen)] = 0;
-    free(raw_digest);
+    digest = strdup(raw_digest);
```
Comment (nrwahl2) [inline lib/common/digest.c:243]: We were asserting before so might as well use `pcmk__str_copy()` now

Comment (clumens) [reply lib/common/digest.c:243]: Ah yeah, I meant to do that.  This is what happens when you eat lunch in the middle of doing something.

Comment (nrwahl2) [reply lib/common/digest.c]: Meh I've done worse

File: lib/common/digest.c [hunk 2]
```diff
@@ -226,37 +226,22 @@ pcmk__xa_filterable(const char *name)
...
+    gchar *raw_digest = NULL;
 
     if (buffer == NULL) {
         return NULL;
     }
 
-    raw_digest = pcmk__assert_alloc(dlen, sizeof(unsigned char));
-
-    rc = gnutls_hash_fast(GNUTLS_DIG_MD5, buffer, strlen(buffer), raw_digest);
+    raw_digest = g_compute_checksum_for_string(G_CHECKSUM_MD5, buffer, strlen(buffer));
```
Comment (kgaillot) [inline lib/common/digest.c:236]: We could just use -1 for the length since it's nul-terminated

**Review by kgaillot — APPROVED** [review]
Yeah glib is a lot cleaner. I don't mind leaving GnuTLS required.

File: lib/common/digest.c (hunk 2 above)
Comment (clumens) [reply lib/common/digest.c]: Fixed.

**Review by kgaillot — APPROVED** [review]

**Pull request merged**